src/BitStringAddresses/bosefs.jl

```diff
@@ -328,14 +328,43 @@ function hopnextneighbour(b::BoseFS{N,M,A}, chosen) where {N,M,A<:BitString}
     end
     return BoseFS{N,M,A}(new_address), √prod
 end
-function hopnextneighbour(b::SingleComponentFockAddress, i)
+function hopnextneighbour(b::SingleComponentFockAddress, i; pitwisted::Bool = false, hardwall::Bool = false)
     src = find_occupied_mode(b, (i + 1) >>> 0x1)
     dst = find_mode(b, mod1(src.mode + ifelse(isodd(i), 1, -1), num_modes(b)))
-
     new_b, val = excitation(b, (dst,), (src,))
-    return new_b, val
+    if pitwisted == true
+        if src.mode==num_modes(b) && dst.mode !=1
```

Comment:
```suggestion
        if src.mode == num_modes(b) && dst.mode !=1
```
It looks better with spaces on both sides of an operator. Apply this elsewhere as well.